src/file_resolver.rs

```diff
@@ -40,25 +40,14 @@ pub struct ResolvedFile {
 impl FileResolver {
     /// Add import path
     pub fn add_import_path(&mut self, path: &Path) -> io::Result<()> {
-        self.import_paths.push((None, path.canonicalize()?));
+        self.import_paths.push((None, path.to_path_buf()));
```

Comment:
To match solc's semantics we don't want to fail if an import path doesn't exist.

I'm happy to depart from this, and I'm also happy to try to canonicalize and just not push a new path when we fail to canonicalize. No strong feelings.

Reply:
The idea here is that if an import path does not exist, then it unusable. So, it's like an early warning system: your import path is no good, which might help the user if they're getting "file not found" errors. At least, that's the idea.

Reply:
The more I think about it the more I like being permissive (but I think we should add a warning).

If humans were the only ones to use compilers I think that a buggy configuration should be a hard fail. But many (most?) compilation invocations originate from a build system, and if there is a bug in the build system and we hard fail when an extra path is passed in, then the user is just stuck w/ no recourse. I think that maybe a good philosophy is be strict with human-facing interfaces and more permissive with black box-facing interfaces?

I'm in a PL lab and I just sent out the bat signal to all the compiler peeps to get their hot takes on best practices for compiler interface behavior

Reply:
General consensus is "do what solc does".

What is best practice to give a warning? Just `eprintln!()`?

Reply:
You can't do a `eprintln!()` in the solang crate, you have to do this in `src/bin/solang.rs`. 

Having said that, `clang` does not complain if a path does not exist either.